From c8a075a8a2719355d27b0743429077fbc1f59365 Mon Sep 17 00:00:00 2001
From: Navneet Barwal <navneet.barwal@iotina.io>
Date: Thu, 18 Aug 2016 15:43:45 +0530
Subject: [PATCH 5/6] added ipv6 addresses for gre tunnel for both secure as
 well as clear SSIDS and both ipv4 and v6 tunel parameters will show up when
 machine is reset until and change is applied to each one of them

---
 .../base-files/files/lib/functions/uci-defaults.sh  | 12 ++++++++++--
 .../skysoft/skysoft_web_admin/cgi/function/wlan.c   | 21 +++++++++------------
 .../skysoft/skysoft_web_admin/cgi/include/uci_cfg.h |  4 ++++
 .../wlan_10_4/net/qca-wifi-10.4/files/qcawifi.sh    |  1 -
 4 files changed, 23 insertions(+), 15 deletions(-)

diff --git a/base/software/common/qsdk/package/base-files/files/lib/functions/uci-defaults.sh b/base/software/common/qsdk/package/base-files/files/lib/functions/uci-defaults.sh
index be1ea74..cb24ef7 100755
--- a/base/software/common/qsdk/package/base-files/files/lib/functions/uci-defaults.sh
+++ b/base/software/common/qsdk/package/base-files/files/lib/functions/uci-defaults.sh
@@ -232,14 +232,18 @@ EOF
 ucidef_set_interface_tap() {
 	uci batch <<EOF
 set network.tap='interface'
+set network.tap.network='commtap'
 set network.tap.proto='gretap'
 set network.tap.mode=1
 set network.tap.ipaddr='11.0.0.1'
 set network.tap.peeraddr='11.0.0.6'
-set network.tap.network='commtap'
 set network.tap.enable_vlan='1'
 set network.tap.ka_pri_addr='11.0.0.6'
 set network.tap.ka_sec_addr='11.0.0.110'
+set network.tap.ip6addr='2001:db9::1'
+set network.tap.peer6addr='2001:db9::2'
+set network.tap.ka6_pri_addr='2001:db9::2'
+set network.tap.ka6_sec_addr='2001:db9::3'
 set network.tap.ka_intrvl='3'
 set network.tap.ka_thrld='3'
 set network.tap.ka_pri_chk_intrvl='3'
@@ -250,14 +254,18 @@ EOF
 ucidef_set_interface_clr_tap() {
 	uci batch <<EOF
 set network.clr_tap='interface'
+set network.clr_tap.network='clr_commtap'
 set network.clr_tap.proto='gretap'
 set network.tap.mode=1
 set network.clr_tap.ipaddr='11.0.0.1'
 set network.clr_tap.peeraddr='11.0.0.60'
-set network.clr_tap.network='clr_commtap'
 set network.clr_tap.enable_vlan='1'
 set network.clr_tap.ka_pri_addr='11.0.0.60'
 set network.clr_tap.ka_sec_addr='11.0.0.11'
+set network.clr_tap.ip6addr='2001:db9::1'
+set network.clr_tap.peer6addr='2001:db9::4'
+set network.clr_tap.ka6_pri_addr='2001:db9::4'
+set network.clr_tap.ka6_sec_addr='2001:db9::5'
 set network.clr_tap.ka_intrvl='3'
 set network.clr_tap.ka_thrld='3'
 set network.clr_tap.ka_pri_chk_intrvl='3'
diff --git a/base/software/common/qsdk/package/skysoft/skysoft_web_admin/cgi/function/wlan.c b/base/software/common/qsdk/package/skysoft/skysoft_web_admin/cgi/function/wlan.c
index 27760a6..aa4d0b0 100755
--- a/base/software/common/qsdk/package/skysoft/skysoft_web_admin/cgi/function/wlan.c
+++ b/base/software/common/qsdk/package/skysoft/skysoft_web_admin/cgi/function/wlan.c
@@ -14052,7 +14052,7 @@ int getspectrumwifi(tcm_session *session, char *name, char *val, char *list)
 	}
 	json_object_object_add(obj, "gre_mode", json_object_new_int(atoi(tmp_char)));
 
-	if(!strcmp(tmp_char, "1")) {
+	if(!(strcmp(tmp_char, "1")) || !(strcmp(tmp_char, "0"))) {
 		ret = do_uci_get(SPEC_KA_PRI_ADDR , tmp_char);
 
 		if(ret != 0) {
@@ -14086,9 +14086,7 @@ int getspectrumwifi(tcm_session *session, char *name, char *val, char *list)
 		}
 		json_object_object_add(obj, "peersecaddrcaptive", json_object_new_string(tmp_char));
 
-	}
-	else {
-		ret = do_uci_get(SPEC_KA_PRI_ADDR , tmp_char);
+		ret = do_uci_get(SPEC_KA6_PRI_ADDR , tmp_char);
 
 		if(ret != 0) {
 			cgi_err("get SPEC remote ipaddr err\n");
@@ -14096,7 +14094,7 @@ int getspectrumwifi(tcm_session *session, char *name, char *val, char *list)
 		}
 		json_object_object_add(obj, "peeraddr6", json_object_new_string(tmp_char));
 
-		ret = do_uci_get(SPEC_KA_SEC_ADDR , tmp_char);
+		ret = do_uci_get(SPEC_KA6_SEC_ADDR , tmp_char);
 
 		if(ret != 0) {
 			cgi_err("get SPEC remote secondary ipaddr err\n");
@@ -14106,7 +14104,7 @@ int getspectrumwifi(tcm_session *session, char *name, char *val, char *list)
 
 
 		//captive
-		ret = do_uci_get(SPEC_KA_PRI_ADDR_CAPTIVE , tmp_char);
+		ret = do_uci_get(SPEC_KA6_PRI_ADDR_CAPTIVE , tmp_char);
 
 		if(ret != 0) {
 			cgi_err("get SPEC remote ipaddr captive err\n");
@@ -14114,7 +14112,7 @@ int getspectrumwifi(tcm_session *session, char *name, char *val, char *list)
 		}
 		json_object_object_add(obj, "peeraddr6captive", json_object_new_string(tmp_char));
 
-		ret = do_uci_get(SPEC_KA_SEC_ADDR_CAPTIVE , tmp_char);
+		ret = do_uci_get(SPEC_KA6_SEC_ADDR_CAPTIVE , tmp_char);
 
 		if(ret != 0) {
 			cgi_err("get SPEC remote secondary ipaddr captive err\n");
@@ -14378,7 +14376,6 @@ int setspectrumwifi(tcm_session *session, char *name, char *val, char *list)
         do_cmd("uci delete network.clr_com");
         do_cmd("uci set network.clr_com=interface");
         do_cmd("uci set network.clr_com.type='bridge'");
-        do_cmd("uci set network.clr_com.proto='static'");
 
         do_cmd("uci delete network.clr_commtap");
         do_cmd("uci set network.clr_commtap=interface");
@@ -14685,26 +14682,26 @@ int setspectrumwifi(tcm_session *session, char *name, char *val, char *list)
 		json_object_object_get_ex(i_obj, "peeraddr6", &sub_obj);
 		if(sub_obj) {
 			do_uci_set(SPEC_PERR_IP6, (char*)json_object_get_string(sub_obj));
-			do_uci_set(SPEC_KA_PRI_ADDR , (char*)json_object_get_string(sub_obj));
+			do_uci_set(SPEC_KA6_PRI_ADDR , (char*)json_object_get_string(sub_obj));
 			do_uci_set(QOS_SPEC_PERR_IP, (char*)json_object_get_string(sub_obj));
 
 		}
 
 		json_object_object_get_ex(i_obj, "peersecaddr6", &sub_obj);
 		if(sub_obj) {
-			do_uci_set(SPEC_KA_SEC_ADDR , (char*)json_object_get_string(sub_obj));
+			do_uci_set(SPEC_KA6_SEC_ADDR , (char*)json_object_get_string(sub_obj));
 		}
 
 		json_object_object_get_ex(i_obj, "peeraddr6captive", &sub_obj);
 		if(sub_obj) {
 			do_uci_set(SPEC_PERR_IP6_CAPTIVE, (char*)json_object_get_string(sub_obj));
-			do_uci_set(SPEC_KA_PRI_ADDR_CAPTIVE , (char*)json_object_get_string(sub_obj));
+			do_uci_set(SPEC_KA6_PRI_ADDR_CAPTIVE , (char*)json_object_get_string(sub_obj));
 
 		}
 
 		json_object_object_get_ex(i_obj, "peersecaddr6captive", &sub_obj);
 		if(sub_obj) {
-			do_uci_set(SPEC_KA_SEC_ADDR_CAPTIVE , (char*)json_object_get_string(sub_obj));
+			do_uci_set(SPEC_KA6_SEC_ADDR_CAPTIVE , (char*)json_object_get_string(sub_obj));
 		}
 
 
diff --git a/base/software/common/qsdk/package/skysoft/skysoft_web_admin/cgi/include/uci_cfg.h b/base/software/common/qsdk/package/skysoft/skysoft_web_admin/cgi/include/uci_cfg.h
index 398678b..420dca0 100644
--- a/base/software/common/qsdk/package/skysoft/skysoft_web_admin/cgi/include/uci_cfg.h
+++ b/base/software/common/qsdk/package/skysoft/skysoft_web_admin/cgi/include/uci_cfg.h
@@ -1033,7 +1033,11 @@
 #define SPEC_KA_PRI_ADDR_CAPTIVE			"network.clr_tap.ka_pri_addr"
 #define SPEC_KA_SEC_ADDR_CAPTIVE			"network.clr_tap.ka_sec_addr"
 
+#define SPEC_KA6_PRI_ADDR			"network.tap.ka6_pri_addr"
+#define SPEC_KA6_SEC_ADDR			"network.tap.ka6_sec_addr"
 
+#define SPEC_KA6_PRI_ADDR_CAPTIVE			"network.clr_tap.ka6_pri_addr"
+#define SPEC_KA6_SEC_ADDR_CAPTIVE			"network.clr_tap.ka6_sec_addr"
 
 #define SPEC_KA_INTRVL			"network.tap.ka_intrvl"
 #define SPEC_KA_THRLD			"network.tap.ka_thrld"
diff --git a/base/software/common/qsdk/qca/feeds/wlan_10_4/net/qca-wifi-10.4/files/qcawifi.sh b/base/software/common/qsdk/qca/feeds/wlan_10_4/net/qca-wifi-10.4/files/qcawifi.sh
index 2e44894..d2660b8 100755
--- a/base/software/common/qsdk/qca/feeds/wlan_10_4/net/qca-wifi-10.4/files/qcawifi.sh
+++ b/base/software/common/qsdk/qca/feeds/wlan_10_4/net/qca-wifi-10.4/files/qcawifi.sh
@@ -2144,7 +2144,6 @@ config wifi-iface '${spectrumwifi}_clear'
 	option maxsta		100
 	option rrm		1
 	option ieee80211w 	0
-	option ownip		'14.0.0.50'
 	option rsn_preauth	1
 	option lan_restricted	0
 	option ieee80211r       '0'
-- 
1.9.1

